examples/texture-arrays/main.rs

```diff
@@ -0,0 +1,323 @@
+#[path = "../framework.rs"]
+mod framework;
+
+use bytemuck::{Pod, Zeroable};
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+struct Vertex {
+    _pos: [f32; 2],
+    _tex_coord: [f32; 2],
+    _index: u32,
+}
+
+unsafe impl Pod for Vertex {}
+unsafe impl Zeroable for Vertex {}
+
+fn vertex(pos: [i8; 2], tc: [i8; 2], index: i8) -> Vertex {
+    Vertex {
+        _pos: [pos[0] as f32, pos[1] as f32],
+        _tex_coord: [tc[0] as f32, tc[1] as f32],
+        _index: index as u32,
+    }
+}
+
+fn create_vertices() -> Vec<Vertex> {
+    vec![
+        // left rectangle
+        vertex([-1, -1], [0, 1], 0),
+        vertex([-1, 1], [0, 0], 0),
+        vertex([0, 1], [1, 0], 0),
+        vertex([0, -1], [1, 1], 0),
+        // right rectangle
+        vertex([0, -1], [0, 1], 1),
+        vertex([0, 1], [0, 0], 1),
+        vertex([1, 1], [1, 0], 1),
+        vertex([1, -1], [1, 1], 1),
+    ]
+}
+
+fn create_indices() -> Vec<u16> {
+    vec![
+        // Left rectangle
+        0, 1, 2, // 1st
+        2, 0, 3, // 2nd
+        // Right rectangle
+        4, 5, 6, // 1st
+        6, 4, 7, // 2nd
+    ]
+}
+
+#[derive(Copy, Clone)]
+enum Color {
+    RED,
+    GREEN,
+}
+
+fn create_texture_data(color: Color) -> Vec<u8> {
+    let mut data: Vec<u8> = match color {
+        Color::RED => vec![255, 0, 0, 255],
+        Color::GREEN => vec![0, 255, 0, 255],
+    };
+
+    // Pad data out to require bytes per row alignment
```

Comment:
This isn't needed at all for `write_texture` - the benefit of a built-in function. To clarify, since `write_data` does the copy into the staging area, it takes responsibility on aligning the rows properly for you.

Reply:
What should the bytes_per_row be set to for write_texture, the actual value (in this case 4)?

Reply:
yes, the actual value of 4